src/pymatgen/io/vasp/outputs.py

```diff
@@ -2323,7 +2326,7 @@ def read_table_pattern(
         attribute_name: str | None = None,
         last_one_only: bool = True,
         first_one_only: bool = False,
-    ) -> list:
+    ) -> list:  # TODO: clarify table-like data type
```

Comment:
Tag: important: clarify type of table-like data

Reply:
I would clarify the docstring instead, as the full type of a table is super long (`list[list[Any] | dict[str, Any]]`) and I don't want to add a type alias as it's used only once.

Also the type could vary if last_one_only or first_one_only is True, which would require writing 3 overloads